### PR TITLE
doc: Remove zlib from the list of build dependencies

### PR DESCRIPTION
**Pull request by gvancuts**
Remove zlib from the list of build dependencies and correct an error regarding the Ubuntu 16.04 build environment (which does *not* require to update `gcc` and `binutils`

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by dbkinder** [comment]
@NanlinXie is the change from Ubuntu 18 to 16 OK?

**Comment by NanlinXie** [comment]
@gvancuts , we want to use Ubuntu 18.04 as our Ubuntu guide basic.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by gvancuts** [comment]
I have changed the note to make it more generic, can you please take another look @NanlinXie @dbkinder ?

File: doc/getting-started/building-from-source.rst [hunk 1]
```diff
@@ -92,27 +92,13 @@ Install the necessary tools for the following systems:
...
-        $ sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 60 \
-                             --slave /usr/bin/g++ g++ /usr/bin/g++-7
-
-
-     ACRN development requires ``binutils`` version 2.27 (or higher).
-     Verify your version of ``binutils`` with the command ``apt show binutils``.
-
+     ACRN requires ``gcc`` version 7.3.* (or higher) and ``binutils`` version
+     2.27 (or higher). Check your development environment to ensure you have
+     appropriate versions of these packages.
```
Comment (dbkinder) [inline doc/getting-started/building-from-source.rst:101]: How about we make it easy on the reader and show them how to see what the version numbers are for these tools.  Add:

```
by using the commands: ``gcc -v`` and ``ld -v``.
```

Comment (gvancuts) [reply doc/getting-started/building-from-source.rst]: Good idea, done!

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**